modules/message-lane/src/lib.rs

```diff
@@ -335,22 +335,22 @@ decl_module! {
 
 				// reward relayers that have delivered messages
 				// this loop is bounded by `T::MaxUnconfirmedMessagesAtInboundLane` on the bridged chain
-				for (nonce, relayer) in lane_data.relayers {
-					// we may receive the same confirmation multiple times
-					if nonce < received_range.0 {
-						continue;
-					}
-
-					let message_data = OutboundMessages::<T, I>::get(MessageKey {
-						lane_id,
-						nonce,
-					}).expect("message was just confirmed; we never prune unconfirmed messages; qed");
+				for (nonce_low, nonce_high, relayer) in lane_data.relayers {
+					let nonce_begin = sp_std::cmp::max(nonce_low, received_range.0);
+					let nonce_end = sp_std::cmp::min(nonce_high, received_range.1);
+					// loop won't proceed if current entry is ahead of received range (begin > end).
+					for nonce in nonce_begin..nonce_end + 1 {
```

Comment:
This may panic if there is no overlap between `(nonce_low, nonce_high)` and `received_range`. Do we validate somewhwre that we won't end up with `nonce_begin > nonce_end`?

Reply:
wdym? `for _ in 200..100 { }` isn't panicking

Reply:
Oh, you're right. I've confused the range itself with slice indexing with an incorrect range. Previous code was not using `received_range.1` though, it seems we didn't filter out correctly or assumed it's done in some other place?

Also I guess it's safe to assume that values are `.0` < `.1`?

Reply:
> Oh, you're right. I've confused the range itself with slice indexing with an incorrect range. Previous code was not using received_range.1 though, it seems we didn't filter out correctly or assumed it's done in some other place?

Yest - I've commented before that using `received_range.1` isn't strictly necessary, because it is already filtered by `InboundLane`. But just in case :)

> Also I guess it's safe to assume that values are .0 < .1?

Yeah - these values are coming from internal code (`InboundLane`). So no need to reverify

Reply:
Yep, we touched on the range handling the `.0 > .1` case in the comments [here](https://github.com/paritytech/parity-bridges-common/pull/418#discussion_r505378092).

Reply:
`nonce_begin..=nonce_end` would have done the same :see_no_evil:

Reply:
As + 1.